### DISCUSSION TITLE
StrongSwan Error

### DISCUSSION THREAD

**Discussion by xAvi98** [strongswan/strongswan#469] [Q&A]

**Comment by Thermi** [comment]
Please follow the guide in #196 (At the very least provide server side logs).

**Reply by xAvi98** [reply]
I readed throught it and i didnt found anything unfortunately ... what do you mean with server side logs? sorry for my english :(

**Reply by Thermi** [reply]
```
7. Now, you may ask for help. Please write in English only. Do not write your messages in any other language.
Please attach your complete config files (ipsec.conf, strongswan.conf, swanctl.conf etc.) and a complete log file showing the problem.
Please supply text files. Pictures are not useful.
We generally require all of the following from you:
The complete log from daemon start to the point where the problem occurs
The complete configuration (ipsec.conf or swanctl.conf, depending on which configuration backend you are using)
The complete current status of the daemon (ipsec statusall or swanctl -L and swanctl -l)
The complete firewall rules (output of iptables-save and ip6tables-save on Linux, analogously on other operating systems using the corresponding command(s))
The complete contents of all routing tables (output of ip route show table all on Linux, analogously on other operating systems)
The complete overview over all IP addresses (output of ip address on Linux, analogously on other operating systems)
```

In a section you can expand at the bottom of the page:

```
strongswan >= 5.7.0
filelog {
        # since 5.7.0 the path to the log file has to be specified in a separate setting if it contains dots,
        # use an arbitrary name without dots for the section instead of the one given here
        charon-debug-log {
                # this setting is required with 5.7.0 and newer if the path contains dots
                path = /var/log/charon_debug.log
                time_format = %a, %Y-%m-%d, %H:%M:%S
                default = 2
                mgr = 0
                net = 1
                enc = 1
                asn = 1
                job = 1
                ike_name = yes
                append = no
                flush_line = yes
        }
}
```
strongswan < 5.7.0
```
filelog {
        /var/log/charon_debug.log {
                time_format = %a, %Y-%m-%d, %%H:%M:%S
                default = 2
                mgr = 0
                net = 1
                enc = 1
                asn = 1
                job = 1
                ike_name = yes
                append = no
                flush_line = yes
        }
}
```

**Comment by xAvi98** [comment]


**Reply by Thermi** [reply]
Please use the filelogger configuration matching your server side strongSwan version, restart the daemon, reproduce the issue, and share the created log file.

**Reply by xAvi98** [reply]
Could you maybe send me an tutorial how to use the filelogger? Sorry i am new to StrongSwan.

**Reply by Thermi** [reply]
https://wiki.strongswan.org/projects/strongswan/wiki/LoggerConfiguration

**Reply by xAvi98** [reply]
[charon.log](https://github.com/strongswan/strongswan/files/6693610/charon.log)

**Reply by xAvi98** [reply]
there is the log ... i think there are no errors

**Reply by tobiasbrunner** [reply]
The first proposal you configured uses an algorithm that's not actually supported on your system: _chacha20poly1305_ is only provided by the _openssl_ plugin since [5.8.0](https://wiki.strongswan.org/versions/73) and you don't have the _chapoly_ plugin loaded. If that proposal is selected (the message on the client regarding the DH group indicates it is), the key derivation will fail later (you should see something like `ENCRYPTION_ALGORITHM CHACHA20_POLY1305 (key size 256) not supported!` in the log when the client connects).

**Reply by xAvi98** [reply]
yes i looked it up... it is like you wrote it. Now i have to install the chapoly plugion ?

**Reply by Thermi** [reply]
You don't see an error because you didn't reproduce the issue as I wrote.

**Reply by xAvi98** [reply]
@Thermi i posted the error below. I would like to thank you in advance for your help and i Iapologize for the inconvenience.

**Reply by Thermi** [reply]
Good. Even if you install the chapoly plugin you don't know if the kernel supports the algorithm. It's needed for ESP.

**Reply by tobiasbrunner** [reply]
> Now i have to install the chapoly plugion ?

You could. Or you configure an algorithm that's actually supported (e.g. _aes256gcm16_), or you don't configure any proposals explicitly, so the default proposal is used (which includes all supported algorithms that are not very weak).

And as @Thermi said, you will also want to change the _esp_ proposal unless you know the kernel supports ChaCha20/Poly1305.

**Reply by xAvi98** [reply]
Okey really thank you guys. is there any tutorial? 
Unfortunately i dont know how to change proposals ...

**Reply by tobiasbrunner** [reply]
Just edit the config file so it says `ike=aes256gcm16-prfsha512-curve25519,...` instead of `ike=chacha20poly1305-sha512-curve25519-prfsha512,...` (and similar for `esp=`). (Or you can just remove both lines to use the default proposals, although in your strongSwan version, there is no AEAD ESP proposal that includes AES-GCM yet, so if you want to use AES-GCM, you have to configure `esp=` manually.) Where did you get that config from anyway?

**Reply by xAvi98** [reply]
Okey thank you i will try it later have to do other stuff right now. If i have more questions i will write back here if its fine for you.
I got this conig from https://www.digitalocean.com under community and tutorials .. but its on german.
